src/ansys/fluent/visualization/pyvista/pyvista_objects.py

```diff
@@ -35,6 +35,52 @@ def __init__(
         )
 
 
+class Contours:
+    def error_check(self, solver):
+        allowed_fields = (
+            solver.field_data.get_scalar_field_data.field_name.allowed_values()
+        )
+        allowed_surfaces = (
+            solver.field_data.get_scalar_field_data.surface_name.allowed_values()
+        )
+        if self.field not in allowed_fields:
+            raise ValueError(
+                f"{self.field} is not valid field. Valid fields are - {allowed_fields}"
+            )
+        for surface in self.surfaces:
+            if surface not in allowed_surfaces:
+                raise ValueError(
+                    f"{surface} is not valid surface. Valid surfaces are - {allowed_surfaces}"  # noqa: E501
+                )
+
+    def __init__(self, *args):
+        if len(args) == 2:
+            self.field, self.surfaces = args[0], args[1]
+        elif len(args) == 3:
+            self.solver = args[2]
+            self.error_check(self.solver)
+            self.field, self.surfaces = args[0], args[1]
+
+    def draw(self, solver, target):
+        self.error_check(solver)
+        existing_contours = solver.results.graphics.contour.get_object_names()
+        import time
+
+        contour_name = f"Contour_{time.time()}"
```

Comment:
To find an available contour name: check for `Contour_1`, `Contour_2`, ... within the existing names and pick the first name which doesn't exist.

Reply:
It allows to create existing contours again.

![image](https://github.com/ansys/pyfluent-visualization/assets/106588300/c62255f9-4ddc-48f4-bed2-0959aebbed1e)

Reply:
To create a new contour through settings API: `solver.results.graphics.contour["<contour-name>"] = <state-dict>`

Reply:
Done. Thank you.